lib/active_model/serializer/adapter/json_api/association.rb

```diff
@@ -0,0 +1,48 @@
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        class Association
+          def initialize(parent_serializer, serializer, options, links, meta)
+            @object = parent_serializer.object
+            @scope = parent_serializer.scope
+
+            @options = options
+            @data = data_for(serializer, options)
+            @links = links
+                     .map { |key, value| { key => Link.new(parent_serializer, value).as_json } }
+                     .reduce({}, :merge)
+            @meta = meta.respond_to?(:call) ? parent_serializer.instance_eval(&meta) : meta
+          end
+
+          def as_json
```

Comment:
if we follow that pattern, yes. It was always about how we feel about it making sense https://github.com/rails-api/active_model_serializers/pull/1004#discussion_r46783510 and https://github.com/rails-api/active_model_serializers/pull/1482